### ISSUE TITLE
Cortex is not using vendored dependencies in CI

### ISSUE THREAD

**Issue by joe-elliott** [cortexproject/cortex#1710]
During builds Cortex is not using its vendored dependencies.  This probably changed in the migration to 1.13 due to changes in the way it handled go modules:

https://github.com/golang/go/wiki/Modules#recent-changes

> GO111MODULE=auto enables module-mode if any go.mod is found, even inside GOPATH. 

Example circleci output from a recent PR:
```
CGO_ENABLED=0 go build -ldflags "-extldflags \"-static\" -s -w" -tags netgo -o cmd/cortex/cortex ./cmd/cortex
go: downloading gopkg.in/yaml.v2 v2.2.2
go: downloading github.com/julienschmidt/httprouter v1.2.0
go: downloading github.com/jonboulle/clockwork v0.1.0
go: downloading github.com/mwitkow/go-conntrack v0.0.0-20190716064945-2f068394615f
go: downloading github.com/hashicorp/consul/api v1.1.0
```

**Comment by bboreham** [comment]
I was under the impression the CI included a check that vendor'd files were the same as the go module files.

**Comment by joe-elliott** [comment]
In the lint workflow it does pull and verify that vendored deps match go modules.  But it redownloads dependencies in every step independently.  So technically the dependencies Cortex is building against are different than the ones that are checked.

Depending on how much you trust the go proxy this may matter less or more, but it's definitely increasing build times to redownload dependencies on every step.

**Comment by bboreham** [comment]
I forget who told me this, but the answer is to add `-mod=vendor` to `go build`.

**Comment by pracucci** [comment]
_There's a PR open for it_

**Comment by pracucci** [comment]
_Still open_

**Comment by pracucci** [comment]
Re-opening cause #2679 doesn't affect unit tests running CI, so it's partially fixed.

**Comment by pracucci** [comment]
Fixed by migration to Golang 1.14.
